### PR TITLE
[IMP] point_of_sale: auto select partner after creation / edition

### PR DESCRIPTION
**Pull request by davidmonnom**
When creating or editing a partner from the POS, the partner is now automatically selected in the partner list.

taskId: 4034971

### CODE REVIEW

File: addons/point_of_sale/static/src/app/store/make_awaitable_dialog.js [hunk 1]
```diff
@@ -18,6 +18,25 @@ export function makeAwaitable(dialog, comp, props, options) {
...
+                onSave: (record) => {
+                    action.doAction({
+                        type: "ir.actions.act_window_close",
+                    });
+                    resolve(record);
+                },
+            },
+        });
+    });
+}
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/store/make_awaitable_dialog.js:38]: Can you find a way to not change the arguments? Maybe just passing the props is enough
```suggestion
export function makeActionAwaitable(action, config, props) {

    return new Promise((resolve) => {
        action.doAction(config, {
            props: {
                ...props,
                onSave: (record) => {
                    action.doAction({
                        type: "ir.actions.act_window_close",
                    });
                    resolve(record);
                },
            },
        });
    });
}
```

**Review by vlst-odoo — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/static/src/app/store/make_awaitable_dialog.js (hunk 1 above)
Comment (davidmonnom) [reply addons/point_of_sale/static/src/app/store/make_awaitable_dialog.js:38]: I need to override onSave anyway to resovle, and you get the value of onSave as the result of the promise. For the rest is standard use

Comment (davidmonnom) [reply addons/point_of_sale/static/src/app/store/make_awaitable_dialog.js]: I just stop to use extra var, but its the same thing.

File: addons/web/static/src/webclient/actions/action_service.js [hunk 2]
```diff
@@ -1312,6 +1312,7 @@ export function makeActionManager(env, router = _router) {
      * @returns {Promise<number | undefined | void>}
      */
     async function doAction(actionRequest, options = {}) {
+        debugger
```
Comment (caburj) [inline addons/web/static/src/webclient/actions/action_service.js:1315]:
```suggestion
```

File: addons/point_of_sale/static/src/app/store/make_awaitable_dialog.js (hunk 1 above)
Comment (caburj) [reply addons/point_of_sale/static/src/app/store/make_awaitable_dialog.js]: I agree with the original. doAction's 2nd args may contain other field that `props`.

**Review by vlst-odoo — APPROVED** [review]
robodoo r+

**Comment by davidmonnom** [comment]
@robodoo r-

**Review by vlst-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**